src/main/java/racingcar/RacingService.java

```diff
@@ -0,0 +1,29 @@
+package racingcar;
+
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class RacingService {
+
+    public void moveByNumOfTriesAndShowResult(Racing racing) {
+        ResultView resultView = new ResultView();
+        for (int i = 0; i < racing.getNumOfTries(); i++) {
+            move(racing);
+            resultView.showCurrentState(racing.getCars());
+        }
+    }
+
+    public void move(Racing racing) {
+        ArrayList<Car> newCars = (ArrayList<Car>) racing.getCars().stream()
+                .map(car -> {
+                    int rand = generateRand();
+                    return rand > racing.getThreshold() ? car.moveFoward(rand) : car;
```

Comment:
문제가 있다기 보다는 객체지향 개발에서 본다면 아쉬움이 있는것 같아요!

`Racing`를 단순한 자료구조로 사용하고 있는것 같아요!
상태를 가진 객체에게 메시지를 전달하고 객체가 직접 상태를 관리할수 있도록 하려면 어떻게해야하는지 고민해보면 좋을것 같아요!

Reply:
저는 정답이 없다고 생각하고, 다양한 관점으로 생각하려고 노력하는 편입니다.

다른 관점으로 본다면, 
```suggestion
                    return racing. forwardable() ? car.moveFoward(rand) : car;
```

`rand > racing.getThreshold()`의 상태 판단을 객체에게 맡김으로써 코드의 의도를 파악하기에 쉬워졌고 응집성이 높아졌다고 생각하기 때문에 생각해보았어요!

Reply:
추가적으로 객체의 메소드가 생겼으므로 TDD를 진행하기에도 더 용이할것 같네요! 😄